File: DigitalRSVP.Core/Models/invitation.cs
using System;

namespace DigitalRSVP.Core.Models
{
    public class Invitation
    {
        private Guid _id;
        public Guid Id
        {
            get
            {
                return this._id;
            }
            set
            {
                this._id = value;
            }
        }

        private Guid _eventId;
        public Guid EventId
        {
            get
            {
                return this._eventId;
            }
            set
            {
                this._eventId = value;
            }
        }

        private string _name;
        public string Name
        {
            get
            {
                return this._name;
            }
            set
            {
                this._name = value;
            }
        }

        private bool _weddingParty;
        public bool WeddingParty
        {
            get
            {
                return this._weddingParty;
            }
            set
            {
                this._weddingParty = value;
            }
        }

        private bool _designatedSeating;
        public bool DesignatedSeating
        {
            get
            {
                return this._designatedSeating;
            }
            set
            {
                this._designatedSeating = value;
            }
        }

        private string? _noteToInvitee;
        public string? NoteToInvitee
        {
            get
            {
                return this._noteToInvitee;
            }
            set
            {
                this._noteToInvitee = value;
            }
        }

        private DateTime _createdDate;
        public DateTime Created_Date
        {
            get
            {
                return this._createdDate;
            }
            set
            {
                if (value > new DateTime(2025, 1, 1))
                {
                    this._createdDate = value;
                }
            }
        }

        private DateTime _updatedDate;
        public DateTime Updated_Date
        {
            get
            {
                return this._updatedDate;
            }
            set
            {
                if (value > new DateTime(2025, 1, 1))
                {
                    this._createdDate = value;
                }
            }
        }
    }
}

File: DigitalRSVP.App/src/app/data/invitation.ts
export class Invitation {
    private _id: string = '';
    private _eventId: string = '';
    private _name: string = '';
    private _weddingParty: boolean = false;
    private _designatedSeating: boolean = false;
    private _noteToInvitee: string = '';
    private _createdDate: Date = new Date();
    private _updatedDate: Date = new Date();

    constructor(id: string) {
        this._id = id;
    }

    get id(): string {
        return this._id;
    }
    set id(value: string) {
        this._id = value;
    }

    get eventId(): string {
        return this._eventId;
    }
    set eventId(value: string) {
        this._eventId = value;
    }

    get name(): string {
        return this._name;
    }
    set name(value: string) {
        this._name = value;
    }

    get weddingParty(): boolean {
        return this._weddingParty;
    }
    set weddingParty(value: boolean) {
        this._weddingParty = value;
    }

    get designatedSeating(): boolean {
        return this._designatedSeating;
    }
    set designatedSeating(value: boolean) {
        this._designatedSeating = value;
    }

    get noteToInvitee(): string {
        return this._noteToInvitee;
    }
    set noteToInvitee(value: string) {
        this._noteToInvitee = value;
    }

    get created_Date(): Date {
        return this._createdDate;
    }
    set created_Date(value: Date) {
        this._createdDate = value;
    }

    get updated_Date(): Date {
        return this._updatedDate;
    }
    set updated_Date(value: Date) {
        this._updatedDate = value;
    }
}